src/components/schedule-page/table/schedule/sections/base-section/use-selection-matrix.ts

```diff
@@ -1,10 +1,14 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 export type SelectionMatrix = boolean[][];
 
+export function areDimesionsEqual(matrix1: unknown[][], matrix2: unknown[][]): boolean {
```

Comment:
@Qwebeck Mógłbyś opisać jaki był problem?

Reply:
Tak, sorki, że nie opisałem wcześniej.

Problem był taki, że w momencie kiedy zmienialiśmy stan grafiku z aktualnego na bazowy, to nie komponent się renderował od nowa, ale macierz się nie odświeżała, i miała mniejszy rozmiar. 
Komponent sekcji jest renderowany w taki sposób
```
 <RowComponent
            selection={[...selectionMatrix[rowInd]]}
```

I pojawiała się sytuacja w której próbowaliśmy zrobić spread na `undeinfed` i mieliśmy bug. 

Ale nie j

> 
> 
> > Jak odtworzyć błąd z selection matrix
> > Na developie:
> > 
> > 1. Wczytać grafik https://drive.google.com/file/d/1Hw6_8m_p-EGSyea8JrDUitxk9hBrda0f/view?usp=sharing
> > 2. Zmienić zespół dowolnego pracownika
> 
> Zrobiłem tak, ale błąd się nie pojawił

Umknął mi jeden punkt:
3. Przełączyć na wersję bazową

Reply:
Tak, sorki, że nie opisałem wcześniej.

Problem był taki, że w momencie kiedy zmienialiśmy stan grafiku z aktualnego na bazowy, to nie komponent się renderował od nowa, ale macierz się nie odświeżała, i miała mniejszy rozmiar. 
Komponent sekcji jest renderowany w taki sposób
```
 <RowComponent
            selection={[...selectionMatrix[rowInd]]}
```

I pojawiała się sytuacja w której próbowaliśmy zrobić spread na `undeinfed` i mieliśmy bug. 

Ale nie jestem pewny czy to jest na rozwiązuje problem, czy to rozwiązanie które wprowadziłem sedno problemu, czy tylko eliminuje jeden z jego przejawów.

Reply:
Oki, to zobaczymy czy będzie to nawracać. Na razie działa lepiej, więc mergujemy.